proxy_http.go

```diff
@@ -315,7 +315,16 @@ func (proxy *proxy) writeResponse(downstream io.Writer, req *http.Request, resp
 		resp = prepareResponse(resp, belowHTTP11)
 		proxy.addIdleKeepAlive(resp.Header)
 	}
-	err := resp.Write(out)
+
+	bout := bufio.NewWriter(out)
+	err := resp.Write(bout)
+	// always try to flush what we have
+	err1 := bout.Flush()
```

Comment:
Interesting. If we flush the buffered writer after every resp.Write, how does it help?

Reply:
I guess it's because` resp.write()` internally calls the `io.Writer` several times. Using buffered writer coalesce them into once - when calling `Flush()`.